tests/datetime.test/t5_01.req

```diff
@@ -0,0 +1,9 @@
+set timezone EST
+select cast(0 as datetime)
+set timezone UTC
+select cast(0 as datetime)
+set timezone EST
+select cast(0 as datetime)
+select cast("1970-01-01T000000.000 MARS" as datetime)
+set timezone EST
```

Comment:
You should drop this SET to prove that the TZ was correctly restored.

Reply:
Gotcha Nirbhay.. This is a great suggestion. Sorry it took me a while to understand your point.